src/Microsoft.AspNetCore.Mvc.Core/Authorization/AuthorizeFilter.cs

```diff
@@ -46,6 +77,12 @@ public AuthorizeFilter(AuthorizationPolicy policy)
                 throw new ArgumentNullException(nameof(context));
             }
 
+            Policy = Policy ?? await AuthorizationPolicy.CombineAsync(PolicyProvider, AuthorizeData);
+            if (Policy == null)
+            {
+                return;
```

Comment:
The existing behavior in AuthorizationApplicationModelProvider allowed null policies, they just skipped adding an AuthorizeFilter in that situation, since there's no way to telling if Combine is going to result in a null policy ahead of time, this seemed to mimic the existing behavior.